src/getThreadPictures.js

```diff
@@ -0,0 +1,53 @@
+"use strict";
+
+var utils = require("../utils");
+var log = require("npmlog");
+
+module.exports = function(defaultFuncs, api, ctx) {
+  return function getThreadPictures(threadID, offset, limit, callback) {
+    if(!callback) {
+      callback = function() {};
+    }
+
+    var form = {
+      'thread_id': threadID,
+      'offset': offset,
+      'limit': limit
+    };
+
+    defaultFuncs
+      .post("https://www.facebook.com/ajax/messaging/attachments/sharedphotos.php", ctx.jar, form)
+      .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+      .then(function(resData) {
+        if (resData.error) {
+          throw resData;
+        }
+        return Promise.all(resData['payload']['imagesData'].map(function(image) {
+          return new Promise(function(resolve, reject) {
+            form = {
+              'thread_id': threadID,
+              'image_id': image.fbid
+            };
+            defaultFuncs
+            .post("https://www.facebook.com/ajax/messaging/attachments/sharedphotos.php", ctx.jar, form)
+            .then(utils.parseAndCheckLogin(ctx.jar, defaultFuncs))
+            .then(function(resData) {
+              if (resData.error) {
+                throw resData;
```

Comment:
Turns out I was complicating it by wrapping it in a new promise (`return new Promise`). Fixed code in commit.